### PR TITLE
bugfix `aws-sso`, `github-oidc-provider`

### LINKED ISSUE #744: Missing "root" alias aws provider definition in aws-sso component
Found a bug? Maybe our [Slack Community](https://slack.cloudposse.com) can help.

[Slack Community](https://slack.cloudposse.com)

## Describe the Bug
All versions of the aws-sso component released from 1.227.0 onwards do not have an `alias = "root"` aws provider defined in providers.tf, which prevents planning and/or applying changes to the component if _any_ sso assignments have been made in the root account.

Prior to the release of version 1.227.0, the aws-sso component's providers.tf declared this additional aws provider definition, which is used to deploy the `sso_account_assignments_root` module in main.tf.

## Expected Behavior

- Given that I have defined sso account assignments in the aws-sso component YAML configuration for the account designated as the root account for my organisation ("core-root"),
- When I plan or apply valid changes made to any of the YAML configuration for the component
- Then the component should plan successfully

## Steps to Reproduce
1. Go to the YAML component configuration for the aws-sso component (either the catalog or where it is used)
2. Add an `account_assignments` block for the org root account, typically `core-root` 
3. Ensure this matches the output from account-map such that the aws-sso local `account_assignments_root`is non-empty
4. Run `atmos terraform plan aws-sso --stack core-gbl-identity`(where this is env hosting the aws-sso component)
5. You should see the error: 
```
Error: missing provider provider["registry.terraform.io/hashicorp/aws"].root
```

## Screenshots
N/A

## Environment (please complete the following information):

Anything that will help us triage the bug will help. Here are some ideas:
 - OS: Any
 - Atmos component version affected >1.227.0

## Additional Context
For reference, the 1.226.0 version of providers.tf which contains the root-aliased provider is [here](https://github.com/cloudposse/terraform-aws-components/blob/1.226.0/modules/aws-sso/providers.tf)

The 1.227.0 versions of providers.tf for which the root-aliased provider is missing is [here](https://github.com/cloudposse/terraform-aws-components/blob/1.227.0/modules/aws-sso/providers.tf)

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
* Bugfixes `filter` depreciation issue via module update to `1.1.1`
* Bugfixes missing `aws.root` provider
* Bugfixes `github-oidc-provider` v1.238.1

## why
* Bugfixes

## references
* https://github.com/cloudposse/terraform-aws-sso/pull/44
* closes #744

### CODE REVIEW

File: modules/aws-sso/providers.root.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  # Profile is deprecated in favor of terraform_role_arn. When profiles are not in use, terraform_profile_name is null.
+  profile = module.iam_roles.terraform_profile_name
+
+  dynamic "assume_role" {
+    # module.iam_roles.terraform_role_arn may be null, in which case do not assume a role.
+    for_each = compact([module.iam_roles.terraform_role_arn])
+    content {
+      role_arn = module.iam_roles.terraform_role_arn
+    }
+  }
```
Comment (milldr) [inline modules/aws-sso/providers.root.tf:14]: do we need all of this? The old root.aws provider only had alias and region
```suggestion
```

Comment (Benbentwo) [reply modules/aws-sso/providers.root.tf:14]: https://cloudposse.slack.com/archives/CA4TC65HS/p1688064489135749

**Review by Nuru — CHANGES_REQUESTED** [review]
~Do not re-introduce providers.root. Instead, delete module "sso_account_assignments_root"~

Sorry, my mistake. See #744. 

I missed that `providers.root` is needed because [the delegated account cannot provision permission sets in the org management account](https://docs.aws.amazon.com/singlesignon/latest/userguide/delegated-admin.html#delegated-admin-tasks-member-account). While this makes me question the usefulness of delegating SSO to the `identity` account, it seems clear we are going to need it if we do.

File: modules/aws-sso/main.tf [hunk 2]
```diff
@@ -19,24 +19,12 @@ module "permission_sets" {
...
-  providers = {
-    aws = aws.root
-  }
-
-  account_assignments = local.account_assignments_root
-  context             = module.this.context
-}
-
 locals {
   enabled = module.this.enabled
```
Comment (milldr) [inline modules/aws-sso/main.tf:29]: we'll also want to remove the locals starting with `local.account_assignments_root` and related locals

Comment (Nuru) [reply modules/aws-sso/main.tf]: My mistake. We need the `root` provider and locals, and I have restored them.

**Review by Nuru — APPROVED** [review]

**Pull request merged**